### PR TITLE
Tma 475 address cri tx ma interaction 01

### PR DESCRIPTION
**Pull request by dthampi**
This is the test for Address CRI to TxMA integration to run in staging environment.

### CODE REVIEW

File: tests/audit-tests/src/test/resources/features/FirehoseToS3.feature [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@build @dev @staging
+@build @dev #@staging
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/resources/features/FirehoseToS3.feature:1]:
```suggestion
@build @dev @staging
```
This should be unhashed

File: tests/audit-tests/null/result.json [hunk 2]
```diff
@@ -0,0 +1 @@
...CRI Integration displays the user's address in JSON","match":{"location":"uk.gov.di.txma.audit.bdd.AddressCRI_TxmaStepDefinitions.response_from_Address_CRI_Integration_displays_the_user_s_address_in_JSON()"},"keyword":"Then "},{"result":{"duration":32146000,"status":"passed"},"line":11,"name":"the user clicks on summaryTest and reads the sub value from JSON","match":{"location":"uk.gov.di.txma.audit.bdd.AddressCRI_TxmaStepDefinitions.the_User_Clicks_On_Summary_Test_And_Reads_The_Sub_Value_From_JSON()"},"keyword":"When "},{"result":{"duration":36205689000,"status":"passed"},"line":12,"name":"the audit S3 should have a new event with the postcode provided","match":{"location":"uk.gov.di.txma.audit.bdd.AddressCRI_TxmaStepDefinitions.the_Audit_S3_Should_Have_A_New_Event_With_The_Postcode_Provided()"},"keyword":"Then "}],"tags":[{"name":"@staging"}]}],"name":"Whenever user creates an event on Address CRI UI, TxMA S3 bucket should how a log of this event within a minute.","description":"","id":"whenever-user-creates-an-event-on-address-cri-ui--txma-s3-bucket-should-how-a-log-of-this-event-within-a-minute.","keyword":"Feature","uri":"classpath:features/AddressCRI_Txma.feature","tags":[]}]
```
Comment (Aidan23Smith) [inline tests/audit-tests/null/result.json:1]: Both these null files should be removed

File: tests/audit-tests/null/result.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (Aidan23Smith) [inline tests/audit-tests/null/result.xml:1]: Again, to be removed

File: tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/AddressCRI_TxmaStepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    @Then("Response from Address CRI Integration displays the user's address in JSON")
+    public void response_from_Address_CRI_Integration_displays_the_user_s_address_in_JSON() throws JsonProcessingException {
+        new VerifiableCredentialsPage().ResponseFromAddressCRIIntegration.click();
+        String AddressCRIJSONResponse = new VerifiableCredentialsPage().AddressCRIJSONResponse.getText();
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode = objectMapper.readTree(AddressCRIJSONResponse);
+        sub = jsonNode.get("sub").asText();
+        System.out.println("sub =" +sub);
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/AddressCRI_TxmaStepDefinitions.java:95]: Do you want this output?

File: tests/audit-tests/src/test/java/utilities/Driver.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                        throw new WebDriverException("Your OS doesn't support Safari");
+                    WebDriverManager.getInstance(SafariDriver.class).setup();
+                    driverPool.set(new SafariDriver());
+                    break;
+            }
+        }
+        return driverPool.get();
+    }
+
+    public static void closeDriver() {
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/Driver.java:85]: This is not used. But do you want to use it?

File: tests/audit-tests/src/test/java/utilities/BrowserUtils.java [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+import static org.junit.jupiter.api.Assertions.fail;
+
+
+public class BrowserUtils {
+
+    /**
+     * Switches to new window by the exact title. Returns to original window if target title not found
+     * @param targetTitle
+     */
+    public static void switchToWindow(String targetTitle) {
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/BrowserUtils.java:30]: Most functions in BrowserUtils.java aren't used (so you can remove them)

File: tests/audit-tests/null/result.json (hunk 2 above)
Comment (dthampi) [reply tests/audit-tests/null/result.json:1]: These files comeback automatically after every run even if we remove it. Not sure if I am missing something.

File: tests/audit-tests/null/result.xml (hunk 3 above)
Comment (dthampi) [reply tests/audit-tests/null/result.xml:1]: Same as above. These files come back after every run

File: tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/AddressCRI_TxmaStepDefinitions.java (hunk 4 above)
Comment (dthampi) [reply tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/AddressCRI_TxmaStepDefinitions.java:95]: I have removed this now

File: tests/audit-tests/src/test/java/utilities/Driver.java (hunk 5 above)
Comment (dthampi) [reply tests/audit-tests/src/test/java/utilities/Driver.java:85]: Again not sure if I am missing something. Removing this is giving a compiler error

File: tests/audit-tests/src/test/java/utilities/BrowserUtils.java (hunk 6 above)
Comment (dthampi) [reply tests/audit-tests/src/test/java/utilities/BrowserUtils.java:23]: Done

**Pull request merged**